baybe/simulation/lookup.py

```diff
@@ -1,201 +1,301 @@
-        #   a dataframe with properly labeled columns.
-
-        # Since the return of a lookup function is a tuple, the following code stores
-        # tuples of floats in a single column with label 0:
-        measured_targets = queries.apply(lambda x: lookup(*x.values), axis=1).to_frame()
-        # We transform this column to a DataFrame in which there is an individual
-        # column for each of the targets....
-        split_target_columns = pd.DataFrame(
-            measured_targets[0].to_list(), index=measured_targets.index
+    # Since the return of a lookup function is a tuple, the following code stores
+    # tuples of floats in a single column with label 0:
+    measured_targets = queries.apply(lambda x: lookup(*x.values), axis=1).to_frame()
+    # We transform this column to a DataFrame in which there is an individual
+    # column for each of the targets....
+    split_target_columns = pd.DataFrame(
+        measured_targets[0].to_list(), index=measured_targets.index
+    )
+    # ... and assign this to measured_targets in order to have one column per target
+    measured_targets[split_target_columns.columns] = split_target_columns
+    if measured_targets.shape[1] != len(targets):
+        raise AssertionError(
+            "If you use an analytical function as lookup, make sure "
+            "the configuration has the right amount of targets "
+            "specified."
         )
-        # ... and assign this to measured_targets in order to have one column per target
-        measured_targets[split_target_columns.columns] = split_target_columns
-        if measured_targets.shape[1] != len(campaign.targets):
-            raise AssertionError(
-                "If you use an analytical function as lookup, make sure "
-                "the configuration has the right amount of targets "
-                "specified."
+    for k_target, target in enumerate(targets):
+        queries[target.name] = measured_targets.iloc[:, k_target]
+
+
+def _look_up_targets_from_dataframe(
+    queries: pd.DataFrame,
+    targets: Collection[Target],
+    lookup: pd.DataFrame,
+    impute_mode: Literal[
+        "error", "worst", "best", "mean", "random", "ignore"
+    ] = "error",
+) -> None:
+    """Look up target values from a dataframe."""
+    # # IMPROVE: Although it's not too important for a simulation, this
+    # #  could also be implemented for approximate matches"""
+
+    target_names = [t.name for t in targets]
+
+    all_match_vals = []
+    for _, row in queries.iterrows():
+        # IMPROVE: to the entire matching at once via a merge
+        ind = lookup[
+            (lookup.loc[:, row.index] == row).all(axis=1, skipna=False)
+        ].index.values
+
+        if len(ind) > 1:
+            # More than two instances of this parameter combination
+            # have been measured
+            _logger.warning(
+                "The lookup rows with indexes %s seem to be "
```

Comment:
Why no `f`-string?

Reply:
i remember writing it like that because the logger doesnt support that at runtime and pylint asked me politely to change it to this version :) will be deleted anyway because it should be a `UserWarning`

Reply:
yes, and as discussed in the meeting: code has not been touched during move, so will resolve (next PR will take care of it)